### PR TITLE
fix(gs): Broadcast by viewports

### LINKED ISSUE #99: Trail can be incorrectly shown or incorrectly hidden

### PR DESCRIPTION
**Pull request by splixcord**
Fixes #99

### CODE REVIEW

File: gameServer/src/gameplay/Game.js [hunk 1]
```diff
@@ -488,7 +488,7 @@ export class Game {
 	broadcastHitLineAnimation(player, hitByPlayer) {
 		const position = hitByPlayer.getPosition();
 		const didHitSelf = player == hitByPlayer;
-		for (const nearbyPlayer of this.getOverlappingViewportPlayersForPos(position)) {
+		for (const nearbyPlayer of this.getOverlappingViewportPlayersForRect(player.getTrailBounds())) {
```
Comment (jespertheend) [inline gameServer/src/gameplay/Game.js:491]: I don't think the hitline and honk are necessary? Since these always happen at a fixed point.

**Review by jespertheend — COMMENTED** [review]
Did you run into any issues when using `player.#inOtherPlayerViewports`?

You could make it public by adding a public generator:
```js
*inOtherPlayerViewports() {
    yield* this.#inOtherPlayerViewports;
}
```

**Comment by splixcord** [comment]
I've made it use `inOtherPlayerViewports` for all broadcasts, except `game.broadcastPlayerState` because of the bug where players joining the game do not receive their own data until a small moment has passed.

`Player.removedFromGame()` can also be moved to a game broadcast.

**Review by jespertheend — APPROVED** [review]
LGTM, thanks!

**Pull request merged**